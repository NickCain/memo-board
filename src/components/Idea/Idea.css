.Idea {
    max-width: 140px;
    height: 140px;
    margin: 10px;
    padding: 5px;
    flex-grow: 1;
    width: 100%;
    outline: 2px solid #000000;
}

.Idea .title {
    cursor: pointer;
}

.Idea .input {
    margin: 10px;
}

.Idea .body {
    cursor: pointer;
    display: flex;
    justify-content: center;
}
