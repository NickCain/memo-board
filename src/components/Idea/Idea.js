import PropTypes from 'prop-types';
import React, { Component } from 'react';

import Input from '../../components/common/Input/Input';
import TextArea from '../../components/common/TextArea/TextArea';

import './Idea.css';

class Idea extends Component {
  constructor(props) {
    super(props);

    this.state = {
      editTitle: true,
      editBody: false
    };
  }

  handleTitleFocus = () => {
    this.setState({
      editTitle: true
    })
  };

  handleTitleInputKeyPress = (e) => {
    if (e.key === 'Enter') {
      this.setState({
        editTitle: false
      });
      this.handleUpdateIdea({ type: 'title', text: e.target.value });
    }
  };

  handleTitleInputOnBlur = (e) => {
    this.setState({
      editTitle: false
    });

    this.handleUpdateIdea({ type: 'title', text: e.target.value });
  };

  handleBodyFocus = () => {
    this.setState({
      editBody: true
    })
  };


  handleBodyInputOnBlur = (e) => {
    this.setState({
      editBody: false
    });

    this.handleUpdateIdea({ type: 'body', text: e.target.value });
  };

  handleBodyInputKeyPress = (e) => {
    if (e.key === 'Enter') {
      this.setState({
        editBody: false
      });
      this.handleUpdateIdea({ type: 'body', text: e.target.value });
    }
  };

  handleUpdateIdea = (update) => {
    const { id, createdDate, title, body } = this.props;

    this.props.onUpdateIdea({
      id,
      createdDate,
      title: update.type === 'title' ? update.text : title,
      body: update.type === 'body' ? update.text : body,
    });
  };

  render() {
    const { title, body } = this.props;
    const { editTitle, editBody } = this.state;
    return (
      <div className="Idea">
        {
          editTitle ?
            <Input
              autoFocus
              onBlur={this.handleTitleInputOnBlur}
              onKeyPress={this.handleTitleInputKeyPress}
              placeholder="Type title..."
              text={title}
            />
            :
            <p className="title" onClick={this.handleTitleFocus}>
              {title}
            </p>

        }
        {
          editBody ?
            <TextArea
              autoFocus
              placeholder="type body..."
              onKeyPress={this.handleBodyInputKeyPress}
              onBlur={this.handleBodyInputOnBlur}
              text={body}
            />
            :
            <div
              className="body"
              onClick={this.handleBodyFocus}
            >
              <p className="body-content">
                {body}
              </p>
            </div>
        }
      </div>
    );
  }
}

Idea.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string
};

export default Idea;
