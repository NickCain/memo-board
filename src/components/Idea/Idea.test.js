import { mount, shallow } from 'enzyme';
import React from 'react';
import Idea from './Idea.js'

let wrapper;
let shallowWrapper;

describe('Idea - Component', () => {
  describe('Idea renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <Idea />
      );
      shallowWrapper = shallow(
        <Idea />
      );
    });

    it('should render the Idea component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });
  });
});
