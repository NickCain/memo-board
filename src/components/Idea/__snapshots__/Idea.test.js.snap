// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Idea - Component Idea renders should render the Idea component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Idea"
>
    <Input
        autoFocus={true}
        onBlur={[Function]}
        onKeyPress={[Function]}
        placeholder="Type title..."
        text={undefined}
    />
    <div
        className="body"
        onClick={[Function]}
    >
        <p
            className="body-content"
        />
    </div>
</div>,
  "nodes": Array [
    <div
      className="Idea"
>
      <Input
            autoFocus={true}
            onBlur={[Function]}
            onKeyPress={[Function]}
            placeholder="Type title..."
            text={undefined}
      />
      <div
            className="body"
            onClick={[Function]}
      >
            <p
                  className="body-content"
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Idea />,
      "_debugID": 8,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Idea {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleBodyFocus": [Function],
        "handleBodyInputKeyPress": [Function],
        "handleBodyInputOnBlur": [Function],
        "handleTitleFocus": [Function],
        "handleTitleInputKeyPress": [Function],
        "handleTitleInputOnBlur": [Function],
        "handleUpdateIdea": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "editBody": false,
          "editTitle": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="Idea"
>
          <Input
                    autoFocus={true}
                    onBlur={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Type title..."
                    text={undefined}
          />
          <div
                    className="body"
                    onClick={[Function]}
          >
                    <p
                              className="body-content"
                    />
          </div>
</div>,
        "_debugID": 9,
        "_renderedOutput": <div
          className="Idea"
>
          <Input
                    autoFocus={true}
                    onBlur={[Function]}
                    onKeyPress={[Function]}
                    placeholder="Type title..."
                    text={undefined}
          />
          <div
                    className="body"
                    onClick={[Function]}
          >
                    <p
                              className="body-content"
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Idea />,
}
`;
